packages/blockchain-extension/extension/explorer/environmentExplorer.ts

```diff
@@ -209,71 +214,139 @@ export class BlockchainEnvironmentExplorerProvider implements BlockchainExplorer
+                if (environment.environmentType === EnvironmentType.LOCAL_ENVIRONMENT) {
+                    environmentGroups.push([environment]);
+                } else if (environment.environmentType === EnvironmentType.OPS_TOOLS_ENVIRONMENT || environment.environmentType === EnvironmentType.SAAS_OPS_TOOLS_ENVIRONMENT) {
+                    ibmCloudEnvironments.push(environment);
+                } else {
+                    otherEnvironments.push(environment);
+                }
+                continue;
+            }
 
-                        let runtime: LocalEnvironment | ManagedAnsibleEnvironment;
-                        if (environmentEntry.environmentType === EnvironmentType.LOCAL_ENVIRONMENT) {
-                            runtime = await LocalEnvironmentManager.instance().ensureRuntime(environmentEntry.name, undefined, environmentEntry.numberOfOrgs);
-                        } else {
-                            // Managed ansible
-                            runtime = ManagedAnsibleEnvironmentManager.instance().ensureRuntime(environmentEntry.name, environmentEntry.environmentDirectory);
-                        }
+            // Used to check if group exists already
+            const groupIndex: number = environmentGroups.findIndex((group: FabricEnvironmentRegistryEntry[]) => {
+                return group[0].environmentType === environment.environmentType;
+            });
+
+            if (groupIndex !== -1) {
+                // If a group with the same environmentType exists, then push gateway to the group
+                environmentGroups[groupIndex].push(environment);
+            } else { // if there's a local env that group will have been created first, so check for other env types
+                if (environment.environmentType === EnvironmentType.OPS_TOOLS_ENVIRONMENT || environment.environmentType === EnvironmentType.SAAS_OPS_TOOLS_ENVIRONMENT) {
+                    ibmCloudEnvironments.push(environment);
+                } else {
+                    // group environments that aren't local or opstools
+                    otherEnvironments.push(environment);
+                }
+            }
+        }
 
-                        const treeItem: RuntimeTreeItem = await RuntimeTreeItem.newRuntimeTreeItem(this,
-                            runtime.getName(),
-                            environmentEntry,
-                            {
-                                command: ExtensionCommands.CONNECT_TO_ENVIRONMENT,
-                                title: '',
-                                arguments: [environmentEntry]
-                            },
-                            runtime
-                        );
-
-                        const isRunning: boolean = await runtime.isRunning();
-                        if (isRunning) {
-                            treeItem.contextValue = 'blockchain-runtime-item-running';
-                        }
+        if (ibmCloudEnvironments.length > 0) {
```

Comment:
Don't even need to check this, you can just do `environmentGroups.push(... ibmCloudEnvironments, ... otherEnvironments)`